Python/bytecodes.c

```diff
@@ -1218,9 +1334,9 @@ dummy_func(
                 ERROR_IF(true, error);
             }
             if (PyDict_CheckExact(ns))
-                err = PyDict_SetItem(ns, name, v);
+                err = PyDict_SetItem(ns, name, PyStackRef_AsPyObjectNew(v));
```

Comment:
We need a new reference here because there's `DECREF_INPUTS` after.

Reply:
I've renamed the function to make it clearer what it's actually doing. I realised it might've been misinterpreted as Py_newRef when its real name is to convert defered refs to new references.